### PR TITLE
Storage Sync Swagger preview version (prev PR #109

### PR DESCRIPTION
**Pull request by ankushbindlish2**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json [hunk 1]
```diff
@@ -0,0 +1,3232 @@
...
+              "x-ms-request-id": {
+                "description": "request id.",
+                "type": "string"
+              },
+              "x-ms-correlation-request-id": {
+                "description": "correlation request id.",
+                "type": "string"
+              }
+            }
+
```
Comment (annatisch) [inline specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json]: Operation ID should be in the format group_verb. So this should be changed to:
"operationId": "Workflows_abort"

File: specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json [hunk 2]
```diff
@@ -0,0 +1,3232 @@
...
+              "$ref": "#/definitions/BackupRequest"
+            }
+          }
+        ],
+        "tags": [
+          "CloudEndpoint Resource",
+          "Actions",
+          "Backup Restore"
+        ],
+        "operationId": "CloudEndpointPostBackup_Post",
```
Comment (annatisch) [inline specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json]: Operation ID should be in the format `"group_verb"`. So this should be changed to:
`"operationId": "CloudEndpoints_PostBackup"`

File: specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json [hunk 3]
```diff
@@ -0,0 +1,3232 @@
...
+              "$ref": "#/definitions/PreRestoreRequest"
+            }
+          }
+        ],
+        "tags": [
+          "CloudEndpoint Resource",
+          "Actions",
+          "Backup Restore"
+        ],
+        "operationId": "CloudEndpointPreRestore_Post",
```
Comment (annatisch) [inline specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json]: Operation ID should be in the format `"group_verb"`. So this should be changed to something like:
`"operationId": "CloudEndpoints_PreRestore"`

File: specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json [hunk 4]
```diff
@@ -0,0 +1,3232 @@
...
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "tags": [
+          "CloudEndpoint Resource",
+          "Actions",
+          "Backup Restore"
+        ],
+        "operationId": "CloudEndpointRestoreHeatbeat_Post",
```
Comment (annatisch) [inline specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json]: Operation ID should be in the format `"group_verb"`. So this should be changed to something like:
`"operationId": "CloudEndpoints_RestoreHeartbeat"`

File: specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json [hunk 5]
```diff
@@ -0,0 +1,3232 @@
...
+              "$ref": "#/definitions/PostRestoreRequest"
+            }
+          }
+        ],
+        "tags": [
+          "CloudEndpoint Resource",
+          "Actions",
+          "Backup Restore"
+        ],
+        "operationId": "CloudEndpointPostRestore_Post",
```
Comment (annatisch) [inline specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json]: Operation ID should be in the format `"group_verb"`. So this should be changed to something like:
`"operationId": "CloudEndpoints_PostRestore"`

File: specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json [hunk 6]
```diff
@@ -0,0 +1,3232 @@
...
+            "in": "path",
+            "description": "Name of Sync Group resource.",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "tags": [
+          "ServerEndpoint Resource"
+        ],
+        "operationId": "ServerEndpointsGet_ListBySyncGroup",
```
Comment (annatisch) [inline specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json]: Operation ID should be in the format `"group_verb"`. So this should be changed to something like:
`"operationId": "ServerEndpoints_ListBySyncGroup"`

File: specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json [hunk 7]
```diff
@@ -0,0 +1,3232 @@
...
+            "description": "Name of Server Endpoint object.",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "tags": [
+          "ServerEndpoint Resource",
+          "Actions"
+        ],
+        "operationId": "ServerEndpointRecall_Post",
```
Comment (annatisch) [inline specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json]: Operation ID should be in the format `"group_verb"`. So this should be changed to something like:
`"operationId": "ServerEndpoints_Recall"`

File: specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json [hunk 8]
```diff
@@ -0,0 +1,3232 @@
...
+            "in": "path",
+            "description": "Name of Storage Sync Service resource.",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "tags": [
+          "RegisteredServer Resource"
+        ],
+        "operationId": "RegisteredServersGet_ListByStorageSyncService",
```
Comment (annatisch) [inline specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json]: Operation ID should be in the format `"group_verb"`. So this should be changed to something like:
`"operationId": "RegisteredServers_ListByStorageSyncService"`

File: specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json [hunk 9]
```diff
@@ -0,0 +1,3232 @@
...
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "tags": [
+          "StorageSyncServices Resource"
+        ],
+        "operationId": "StorageSyncServiceByResourceGroupGet_list",
```
Comment (annatisch) [inline specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json]: Operation ID should be in the format `"group_verb"`. So this should be changed to something like:
`"operationId": "StorageSyncServices_ListByResourceGroup"`

File: specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json [hunk 10]
```diff
@@ -0,0 +1,3232 @@
...
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "tags": [
+          "StorageSyncServices Resource"
+        ],
+        "operationId": "StorageSyncServiceGet_ListBySubscription",
```
Comment (annatisch) [inline specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json]: Operation ID should be in the format `"group_verb"`. So this should be changed to something like:
`"operationId": "StorageSyncServices_ListBySubscription"`

File: specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json [hunk 11]
```diff
@@ -0,0 +1,3232 @@
...
+            "in": "path",
+            "description": "Name of Storage Sync Service resource.",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "tags": [
+          "SyncGroup Resource"
+        ],
+        "operationId": "GetSyncGroups_ListByStorageSyncService",
```
Comment (annatisch) [inline specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json]: Operation ID should be in the format `"group_verb"`. So this should be changed to something like:
`"operationId": "SyncGroups_ListByStorageSyncService"`

File: specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json [hunk 12]
```diff
@@ -0,0 +1,3232 @@
...
+            "in": "path",
+            "description": "Name of Sync Group resource.",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "tags": [
+          "CloudEndpoint Resource"
+        ],
+        "operationId": "CloudEndpointsGet_ListBySyncGroup",
```
Comment (annatisch) [inline specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json]: Operation ID should be in the format `"group_verb"`. So this should be changed to something like:
`"operationId": "CloudEndpoints_ListBySyncGroup"`

File: specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json [hunk 13]
```diff
@@ -0,0 +1,3232 @@
...
+              "$ref": "#/definitions/BackupRequest"
+            }
+          }
+        ],
+        "tags": [
+          "CloudEndpoint Resource",
+          "Actions",
+          "Backup Restore"
+        ],
+        "operationId": "CloudEndpointPreBackup_Post",
```
Comment (annatisch) [inline specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json]: Operation ID should be in the format `"group_verb"`. So this should be changed to something like:
`"operationId": "CloudEndpoints_PreBackup"`

**Review by annatisch — CHANGES_REQUESTED** [review]

File: specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json (hunk 1 above)
Comment (ankushbindlish2) [reply specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json]: Resolved. Workflows_Abort

File: specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json (hunk 2 above)
Comment (ankushbindlish2) [reply specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json]: Resolved. CloudEndpoints_PostBackup

File: specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json (hunk 3 above)
Comment (ankushbindlish2) [reply specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json]: Resolved CloudEndpoints_PreRestore

File: specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json (hunk 4 above)
Comment (ankushbindlish2) [reply specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json]: Resolved CloudEndpoints_RestoreHeartbeat

File: specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json (hunk 5 above)
Comment (ankushbindlish2) [reply specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json]: CloudEndpoints_PostRestore

File: specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json (hunk 6 above)
Comment (ankushbindlish2) [reply specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json]: resolved

File: specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json (hunk 7 above)
Comment (ankushbindlish2) [reply specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json]: resolved

File: specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json (hunk 8 above)
Comment (ankushbindlish2) [reply specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json]: resolved

File: specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json (hunk 9 above)
Comment (ankushbindlish2) [reply specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json]: resolved

File: specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json (hunk 10 above)
Comment (ankushbindlish2) [reply specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json]: resolved

File: specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json (hunk 11 above)
Comment (ankushbindlish2) [reply specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json]: resolved

File: specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json (hunk 12 above)
Comment (ankushbindlish2) [reply specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json]: resolved

File: specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json (hunk 13 above)
Comment (ankushbindlish2) [reply specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json]: resolved

File: specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (ankushbindlish2) [reply specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json]: "pauseWaitForSyncDrainTimePeriodInSeconds": {
          "type": "integer",
          "description": "Pre Restore pause wait for sync drain time period in seconds."
		  "x-ms-client-name": "pauseWaitForSyncDrainTimePeriodInSeconds"
        }

[the remark this answers is not in the document]
Comment (ankushbindlish2) [reply specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json]: resolved

[the remark this answers is not in the document]
Comment (ankushbindlish2) [reply specification/storagesync/resource-manager/Microsoft.StorageSync/2017-06-05-preview/storagesync.json]: resolved

**Comment by annatisch** [comment]
@ankushbindlish2 - thanks for all the updates! With regards to pausewaitforsyncdraintimeperiodinseconds, the name of the property needs to be the same capitalization as the request over the wire. So something like this:
```
"pausewaitforsyncdraintimeperiodinseconds": {
  "type": "integer",
  "description": "Pre Restore pause wait for sync drain time period in seconds.",
  "x-ms-client-name": "pauseWaitForSyncDrainTimePeriodInSeconds"
}
```

What this does is it will convert the name to the best language-specific capitalization for the generated SDKs. e.g. for Python, the property will be `pause_wait_for_sync_drain_time_period_in_seconds`. But the SDK will automatically convert it into the correct `pausewaitforsyncdraintimeperiodinseconds` when building the request to send over the wire.

**Review by annatisch — APPROVED** [review]

**Pull request merged**